lib/Dialect/OM/Evaluator/Evaluator.cpp

```diff
@@ -107,15 +107,15 @@ circt::om::Evaluator::instantiate(
     Value value = field.getValue();
 
     FailureOr<evaluator::EvaluatorValuePtr> result =
-        evaluateValue(value, actualParams);
+        evaluateValue(value, actualParams, field.getLoc());
     if (failed(result))
       return failure();
 
     fields[name] = result.value();
   }
 
   // Allocate the Object. Further refinement is expected.
-  auto *object = new evaluator::ObjectValue(cls, fields);
+  auto *object = new evaluator::ObjectValue(cls, fields, cls.getLoc());
```

Comment:
This is still using the ClassOp location for the ObjectValue. I think we want the ObjectOp location, which you get on L196 to be used here.